### PR TITLE
prototype for a rag-agent that periodically calls retrieval tasks

### PR DESCRIPTION
**Pull request by mascharkh**

### CODE REVIEW

**Review by mascharkh — COMMENTED** [review]
lgtm

File: tests/test_scheduler_worker.py [hunk 1]
```diff
@@ -131,12 +133,38 @@ def test_generation(scheduler_client, matrix1):
 
     worker_client = WorkerClient(future.hostName, future.port)
 
+    # TODO: create a single test and get the new values of the output
     expected_result = np.array([
         [[0.5671, 1.3624], [0.112, 0.4556]],
         [[1.4104, 0.7489], [0.2594, 0.6226]]
     ])
+    # expected_result = expected_result[0][0][0]
```
Comment (krish8484) [inline tests/test_scheduler_worker.py:140]: remove commented out code

File: tests/test_scheduler_worker.py [hunk 2]
```diff
@@ -223,4 +251,25 @@ def test_worker_fetching_using_future_where_the_result_is_fetchable_but_errored(
...
+            retrieval_task = Task(taskId=f"Retrieval_{i}", taskDefintion="retrieval", taskData=[json.dumps(knowledge).encode(), json.dumps(embedding).encode()])
+            retrieved_output_future = scheduler_client.SubmitTask(retrieval_task)[0]
+            # retrieved_output = np.array(retrieved_output).reshape(2, 2) --> Make sure this happens in retreival task before it returns
+        generation_task = Task(taskId=f"Generation_{i}", taskDefintion="generation", taskData=[[retrieved_output_future]])
+        generation_output_future = scheduler_client.SubmitTask(generation_task)[0]
+
+        worker_client = WorkerClient(generation_output_future.hostName, generation_output_future.port)
+        generation_output = get_result_from_worker(worker_client, generation_output_future) 
+    
+    logging.info(generation_output)
```
Comment (krish8484) [inline tests/test_scheduler_worker.py:274]: can we do an assertion on the output?

File: tests/test_scheduler_worker.py [hunk 3]
```diff
@@ -123,6 +124,7 @@ def test_retrieval(scheduler_client, matrix1, matrix2):
     worker_client = WorkerClient(future.hostName, future.port)
 
     expected_result = [[1, 2], [3, 4]]
+    validate_result(worker_client, future, expected_result)
     # poll_for_result(worker_client, future, expected_result)
```
Comment (krish8484) [inline tests/test_scheduler_worker.py:127]: remove

**Review by mascharkh — COMMENTED** [review]
lgtm

File: tests/test_scheduler_worker.py [hunk 4]
```diff
@@ -223,4 +250,25 @@ def test_worker_fetching_using_future_where_the_result_is_fetchable_but_errored(
     result = get_result_from_worker(worker_client, resultFuture)
 
     assert result.resultStatus == api_pb2.ResultStatus.ERROR
-    assert result.error.errorType == api_pb2.ErrorType.ERRORWHENEXECUTINGTASK
\ No newline at end of file
+    assert result.error.errorType == api_pb2.ErrorType.ERRORWHENEXECUTINGTASK
+
+
+# Exact replication of rag_agent in our dfut model. Not tested yet. 
+# I think generation task should be failing as its already failing in the above generation test.
```
Comment (swarnim-vyas) [inline tests/test_scheduler_worker.py:257]: remove the comments if it is passing?

File: tests/test_scheduler_worker.py [hunk 5]
```diff
@@ -131,12 +132,38 @@ def test_generation(scheduler_client, matrix1):
...
     worker_client = WorkerClient(future.hostName, future.port)
 
+    # TODO: create a single test and get the new values of the output
     expected_result = np.array([
         [[0.5671, 1.3624], [0.112, 0.4556]],
         [[1.4104, 0.7489], [0.2594, 0.6226]]
     ])
+    # expected_result = expected_result[0][0][0]
+
+    # result_obj = get_result_from_worker(worker_client, future)
```
Comment (swarnim-vyas) [inline tests/test_scheduler_worker.py:142]: clean up the commented code?

**Review by aadhithNarayan — APPROVED** [review]

**Review by swarnim-vyas — APPROVED** [review]

File: tests/test_scheduler_worker.py (hunk 4 above)
Comment (mascharkh) [reply tests/test_scheduler_worker.py:257]: done

**Pull request merged**